### DISCUSSION TITLE
Override ModelManager

### DISCUSSION THREAD

**Discussion by eerison** [sonata-project/SonataAdminBundle#7976] [Q&A] [answered]
Hello!

is there some way to override the `getModelManager`?
I need to customize this method when I'm creating or update! I don't want to save in the database, I want to save in other storage like Message queue! 

https://github.com/sonata-project/SonataAdminBundle/blob/4.x/src/Admin/AbstractAdmin.php#L412

Note: I want to do this for one specific admin only!

**Comment by VincentLanglet** [comment, marked as answer]
Instead of overriding the getModelManager, you should configure the modelManager directly on the admin tag
```
  App\Admin\FooAdmin:
    tags:
      - { name: sonata.admin, model_class: App\Entity\Foo, model_manager: App\ModelManager\CustomModelManager }
```

**Comment by eerison** [comment]
Hey @VincentLanglet 

As I just need to override create and update method, then What I need to decorate: [ModelManager](https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/4.x/src/Model/ModelManager.php)

Do you think that make sense we split the [ModelManagerInterface](https://github.com/sonata-project/SonataAdminBundle/blob/4.x/src/Model/ModelManagerInterface.php#L24) in small interfaces like `CreateModelManagerInterface`, `UpdateModelManagerInterface` and so on?

**Reply by VincentLanglet** [reply]
It could be tempting for developpers but
- There is already too many service/interface the abstractAdmin rely on
- most of the methods of the ModelManagerInterface are related (You wouldn't use a doctrine repository to create an entity and another one to update the entity, and another one to find an entity)
- decorating the modelManager works fine. I did it multiple times.

**Comment by eerison** [comment]
Or maybe here: https://github.com/sonata-project/SonataAdminBundle/blob/4.x/src/Admin/AbstractAdmin.php#L405
